### DISCUSSION TITLE
libp2p http webserver?

### DISCUSSION THREAD

**Discussion by 1066JMay96** [libp2p/rust-libp2p#2875] [Q&A] [answered]
Are there any libp2p web servers I  can use? Thank you! PS if you can find an example that uses Yew for the frontend that would be Poggers!

**Comment by mxinden** [comment, marked as answer]
I am not sure I am following. Note that libp2p does not support HTTP today. It can run on top of a WebSocket connection though.

**Comment by thomaseizinger** [comment]
Note that for go, there is `libp2p-http`: https://github.com/libp2p/go-libp2p-http

I think something similar could be built for Rust, although I am not sure which HTTP clients in Rust allow you to swap out the transport layer.

**Reply by umgefahren** [reply]
[Hyper](https://crates.io/crates/hyper) can be used with custom Io. However they rely on `tokio::io::{AsyncRead, AsyncWrite};` instead of libp2p's `futures::{AsyncRead, AsyncWrite};` (as far as I'm aware). That might be actually an argument for #2901.
